hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OzoneAclUtil.java

```diff
@@ -0,0 +1,311 @@
+   * @param parentAcls
+   * @return true if acls inherited DEFAULT acls from parentAcls successfully,
+   * false otherwise.
+   */
+  public static boolean inheritDefaultAcls(List<OzoneAcl> acls,
+      List<OzoneAcl> parentAcls) {
+    List<OzoneAcl> inheritedAcls = null;
+    if (parentAcls != null && !parentAcls.isEmpty()) {
+      inheritedAcls = parentAcls.stream()
+          .filter(a -> a.getAclScope() == DEFAULT)
+          .map(acl -> new OzoneAcl(acl.getType(), acl.getName(),
+              acl.getAclBitSet(), OzoneAcl.AclScope.ACCESS))
+          .collect(Collectors.toList());
+    }
+    if (inheritedAcls != null && !inheritedAcls.isEmpty()) {
+      inheritedAcls.stream().forEach(acl -> addAcl(acls, acl));
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Convert a list of OzoneAclInfo(protoc) to list of OzoneAcl(java).
+   * @param protoAcls
+   * @return list of OzoneAcl.
+   */
+  public static List<OzoneAcl> fromProtobuf(List<OzoneAclInfo> protoAcls) {
+    return protoAcls.stream().map(acl->OzoneAcl.fromProtobuf(acl))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Convert a list of OzoneAcl(java) to list of OzoneAclInfo(protoc).
+   * @param protoAcls
+   * @return list of OzoneAclInfo.
+   */
+  public static List<OzoneAclInfo> toProtobuf(List<OzoneAcl> protoAcls) {
+    return protoAcls.stream().map(acl->OzoneAcl.toProtobuf(acl))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Add an OzoneAcl to existing list of OzoneAcls.
+   * @param acl
+   * @param existingAcls
+   * @return true if current OzoneAcls are changed, false otherwise.
+   */
+  public static boolean addAcl(List<OzoneAcl> existingAcls, OzoneAcl acl) {
+    if (existingAcls == null || acl == null) {
+      return false;
+    }
+
+    for (OzoneAcl a: existingAcls) {
+      if (a.getName().equals(acl.getName()) &&
+          a.getType().equals(acl.getType()) &&
+          a.getAclScope().equals(acl.getAclScope())) {
+        BitSet current = a.getAclBitSet();
+        BitSet original = (BitSet) current.clone();
+        current.or(acl.getAclBitSet());
+        if (current.equals(original)) {
+          return false;
+        }
+        return true;
+      }
+    }
+
+    existingAcls.add(acl);
+    return true;
+  }
+
+  /**
+   * remove OzoneAcl from existing list of OzoneAcls.
+   * @param acl
+   * @return true if current OzoneAcls are changed, false otherwise.
+   */
+  public static boolean removeAcl(List<OzoneAcl> existingAcls, OzoneAcl acl) {
+    if (existingAcls == null || existingAcls.isEmpty() || acl == null) {
+      return false;
+    }
+
+    for (OzoneAcl a: existingAcls) {
+      if (a.getName().equals(acl.getName()) &&
+          a.getType().equals(acl.getType()) &&
+          a.getAclScope().equals(acl.getAclScope())) {
+        BitSet current = a.getAclBitSet();
+        BitSet original = (BitSet) current.clone();
+        current.andNot(acl.getAclBitSet());
+
+        if (current.equals(original)) {
```

Comment:
Can we add some comments which were there in these methods in earlier code. It will be easy to understand code when reading.

Reply:
I don't think document removed code along with new code is a good idea . But I can explain here for code review. 
Previous in OzoneBucketInfo#removeAcl, we use BitSet#and and BitSet#xor() with complicated logics to implement acl remove. Here we use BitSet#andNot to clear matched bits along with separated unit tests.

Reply:
My comment is to add a few comments like this which were there in earlier methods. Which will be help code readability.

        // We need to do "or" before comparision because think of a case like
        // existing acl is 777 and newly added acl is 444, we have already
        // that acl set. In this case if we do direct check they will not
        // be equal, but if we do or and then check, we shall know it
        // has acl's already set or not.